subsys/spm/secure_services.c

```diff
@@ -164,6 +182,16 @@ int spm_firmware_info_nse(uint32_t fw_address, struct fw_info *info)
 		return -EINVAL;
 	}
 
+	/* Ensure that fw_address is within secure area */
+	if (fw_address >=  PM_SPM_ADDRESS + PM_SPM_SIZE) {
```

Comment:
nit: 2 spaces after `>=`

Reply:
This `>` should be `<`, I think?

Reply:
I don't think so. We are ensuring that `fw_address` is within the secure area, which ends at `PM_SPM_ADDRESS + PM_SPM_SIZE`. If the case is that `fw_address` is above this address, then it is not inside the secure area, and we should return an error. 

I might be blind here, but still looks correct to me :smile:

Reply:
Resolving this since we will use the cmse function instead :+1: